enginetest/queries/queries.go

```diff
@@ -1081,11 +1081,54 @@ var QueryTests = []QueryTest{
 	},
 	{
 		Query: "SELECT t.s,i FROM mytable AS t;",
+		Expected: []sql.Row{
```

Comment:
We need to test with materializing relations. GROUP_BY, WINDOW, UNION, SUBQUERY etc can collect unordered results before returning to parent. So for example, we cannot eliminate a sort like SORT-> GROUP_BY->INDEX_SCAN.

Reply:
Another testing format would be checking that the optimization triggers in subquery aliases, subquery expressions, CTEs, recursive CTEs. Ex: `select * from xy where (select u from uv order by u desc limit 1)`